bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/auth/client/oauth2/OAuthError.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.auth.client.oauth2;
+
+/**
+ * This is an exception class for OAUTH specific errors. i.e. The error responses described in the
+ * RFC6749. Do NOT confuse this with Java errors.
+ *
+ * To keep it simple, this exception class serves for both Authorization Request and Authorization Grant
+ * error response
+ *
+ * The field names are kept exactly the same as the specification.
+ * This allows the error responses to be directly deserialized from JSON.
+ *
+ * @author Gary Tse - initial contribution
+ *
+ * @see https://tools.ietf.org/html/rfc6749#section-4.1.2.1
+ * @see https://tools.ietf.org/html/rfc6749#section-5.2
+ */
+public class OAuthError extends Exception {
```

Comment:
Just to remove confusion, I would prefer to rename it to something that ends with "exception"

Reply:
or you could simply create an exception with custom error codes

Reply:
Fix applied.  The "Error" term comes from the RFC.  I do not wih to change that.